src/redux/features/documentProducer/variables/util.ts

```diff
@@ -0,0 +1,432 @@
+import {
+  TableColumn,
+  TableColumnWithValues,
+  TableVariable,
+  TableVariableWithValues,
+  Variable,
+  VariableUnion,
+  VariableWithValues,
+  isTableVariable,
+} from 'src/types/documentProducer/Variable';
+import { VariableValue, VariableValueTableValue, VariableValueValue } from 'src/types/documentProducer/VariableValue';
+
+type StableId = string;
+type VariableId = number;
+
+const findOutdatedVariable = (variableList: VariableUnion[], variable: Variable): VariableUnion | undefined =>
+  variableList.find((_variable) => _variable.id === variable.replacesVariableId);
+
+const findOutdatedVariableValues = (
+  variable: Variable,
+  values: VariableValue[],
+  variableList: VariableUnion[],
+  tableVariable?: TableVariable
+): VariableValue[] => {
+  if (!variable.replacesVariableId) {
+    return [];
+  }
+
+  let outdatedVariable: VariableUnion | undefined;
+
+  if (tableVariable) {
+    const outdatedTableVariable = findOutdatedVariable(variableList, tableVariable);
+    if (isTableVariable(outdatedTableVariable)) {
+      outdatedVariable = findOutdatedVariable(
+        outdatedTableVariable?.columns.map((column) => column.variable as Variable),
+        variable
+      );
+    }
+  } else {
+    outdatedVariable = findOutdatedVariable(variableList, variable);
+  }
+
+  if (!outdatedVariable) {
+    return [];
+  }
+
+  const outdatedVariableValues: VariableValue[] = values.filter(
+    (val: VariableValue) => val.variableId === outdatedVariable.id
+  );
+
+  return [
+    ...outdatedVariableValues,
+    ...findOutdatedVariableValues(outdatedVariable, values, variableList, tableVariable),
+  ];
+};
+
+export const associateNonSectionVariableValues = (
```

Comment:
This function should probably be put back to how it is in `main`, within the `variablesSelectors.ts` file. A new function was created (`associateNonSectionVariableValues V2`) that splits up the combining of variables and values into different variable types. Only the preliminary work for the `Table` type variable has been completed, this is the most complex one.